### DISCUSSION TITLE
project build failed

### DISCUSSION THREAD

**Discussion by zhongyi5178** [shadowsocks/shadowsocks-android#2918] [General]
mac上 rustup版本信息：

% rustup -V      
rustup 1.24.3 (ce5817a94 2021-05-31)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.62.0 (a8314ef7d 2022-06-27)`

mac上使用android studio编译，log如下：
-------------------------------------------------------------------------------

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/zhongyi/.cargo/registry/src/github.com-1ecc6299db9ec823/chacha20-0.8.1/src/lib.rs:82:5
   |
82 |     feature(stdsimd, aarch64_target_feature)
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: the feature `aarch64_target_feature` has been stable since 1.61.0 and no longer requires an attribute to enable

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /Users/zhongyi/.cargo/registry/src/github.com-1ecc6299db9ec823/chacha20-0.8.1/src/lib.rs:82:13
   |
82 |     feature(stdsimd, aarch64_target_feature)
   |             ^^^^^^^

For more information about this error, try `rustc --explain E0554`.
error: could not compile `chacha20` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
could not compile `chacha20` due to 2 previous errors

build failed, waiting for other jobs to finish...

> Task :core:cargoBuildArm64 FAILED

Execution failed for task ':core:cargoBuildArm64'.
> Process 'command 'cargo'' finished with non-zero exit value 101

**Comment by a451492549** [comment]
I also encountered the same problem. Have you solved it?

**Comment by zhongyi5178** [comment]
> 

回退到更新rule的commit之前

**Comment by a451492549** [comment]
谢谢，运行成功！
